### PR TITLE
fix(ci): usage of `a deprecated Node.js version`

### LINKED ISSUE #21209: `integration.yml` uses `a deprecated Node.js version`
https://github.com/vectordotdev/vector/actions/runs/10709556389

### PR DESCRIPTION
**Pull request by hamirmahal**
fixes #21209.

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -91,7 +91,7 @@ jobs:
       )
     timeout-minutes: 90
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
```
Comment (hamirmahal) [inline .github/workflows/integration.yml:94]: This is the latest major version at the time of this writing.

**Review by jszwedko — COMMENTED** [review]
/ci-run-integration

**Comment by jszwedko** [comment]
Thanks @hamirmahal . I kicked off the tests here.

**Comment by hamirmahal** [comment]
It looks like the 2 failing checks are

https://github.com/vectordotdev/vector/actions/runs/10725231196/job/29742601400?pr=21210
https://github.com/vectordotdev/vector/actions/runs/10725231196/job/29742609937?pr=21210

and they both fail with `Error: Input required and not supplied: app_id`.

**Comment by jszwedko** [comment]
> It looks like the 2 failing checks are
> 
> https://github.com/vectordotdev/vector/actions/runs/10725231196/job/29742601400?pr=21210 https://github.com/vectordotdev/vector/actions/runs/10725231196/job/29742609937?pr=21210
> 
> and they both fail with `Error: Input required and not supplied: app_id`.

Thanks @hamirmahal . I realized I don't think I can trigger those workflows on contributor PRs. I'll mark this to merge and they will run as part of the merge queue.

**Review by jszwedko — APPROVED** [review]

**Comment by jszwedko** [comment]
Thanks again @hamirmahal !

**Comment by hamirmahal** [comment]
You're welcome, @jszwedko!

**Pull request merged**